numpy/__init__.pxd

```diff
@@ -1018,6 +1024,91 @@ cdef extern from *:
     """
 
 
+cdef type floating = <type>(&PyFloatingArrType_Type)
+cdef type integer = <type>(&PyIntegerArrType_Type)
+cdef type complexfloating = <type>(&PyComplexFloatingArrType_Type)
+cdef type bool_ = <type>(&PyBoolArrType_Type)
```

Comment:
These look like they should go into `numpy/__init__.pxd`, no?

Or rather the corresponding type declarations, as @eric-wieser pointed out:
```cython
ctypedef class numpy.integer [object PyObject, type PyIntegerArrType_Type]:
    pass
```

Reply:
> These look like they should go into numpy/__init__.pxd, no?

This is `numpy/__init__.pxd`.

> Or rather the corresponding type declarations, as @eric-wieser pointed out: [...]

I haven't been able to get this to work, either here or implemented directly in the pandas code.  So I just reverted to the known-working implementation.

Exposing integer/floating/complexfloating/bool_ aliases can be done as a follow-up

Reply:
> Exposing integer/floating/complexfloating/bool_ aliases can be done as a follow-up

Sounds good. Let's get this to a point where we can commit it as-is and then iterate on it.

Reply:
> This is numpy/__init__.pxd.

Ah, yes, sorry, I was jumping from project to project here and mixed them up.

> I haven't been able to get this to work

How/where is it failing?